### PR TITLE
capz: add additional presets to conformance jobs

### PR DESCRIPTION
**Pull request by chewong**
Signed-off-by: Ernest Wong <chuwon@microsoft.com>

Replacing `preset-azure-cred` with `preset-azure-cred-only` and `preset-azure-anonymous-pull` to prevent `AZURE_SSH_PUBLIC_KEY_FILE` from being overwritten 

/assign @CecileRobertMichon

### CODE REVIEW

File: config/jobs/kubernetes/sig-cloud-provider/azure/release-master.yaml [hunk 1]
```diff
@@ -517,7 +517,8 @@ periodics:
   labels:
     preset-dind-enabled: "true"
     preset-kind-volume-mounts: "true"
-    preset-azure-cred: "true"
+    preset-azure-cred-only: "true"
+    preset-azure-anonymous-pull: "true"
```
Comment (CecileRobertMichon) [inline config/jobs/kubernetes/sig-cloud-provider/azure/release-master.yaml:521]: right now we're using a local registry for these jobs, do we need the external registry? Are we building k8s artifacts for these ones?

Comment (CecileRobertMichon) [reply config/jobs/kubernetes/sig-cloud-provider/azure/release-master.yaml:521]: If you just want to remove the SSH var you can use `preset-azure-cred-only` without `preset-azure-anonymous-pull`

Comment (chewong) [reply config/jobs/kubernetes/sig-cloud-provider/azure/release-master.yaml]: Switched back to local registry

**Pull request merged**